lib/SauceLabs.js

```diff
@@ -217,13 +219,42 @@ SauceLabs.prototype.createPublicLink = function (id, date, useHour, callback) {
   callback(null, link);
 };
 
+SauceLabs.prototype.uploadFile = function (file, overwrite, callback) {
+  var self = this,
+      filename = path.basename(file);
+
+  fs.readFile(file, function (err, data) {
+    if (err) {
+      callback(err);
+    } else {
+
+      self.send({
+        method: 'POST',
+        path: 'storage/:username/:filename',
+        args: {filename: filename},
+        query: {overwrite: overwrite},
+        uploadData: data
+
+      }, callback);
+    }
+  });
+};
+
+SauceLabs.prototype.getStoredFiles = function (callback) {
+  this.send({
+    method: 'GET',
+    path: 'storage/:username'
+  }, callback);
+};
+
 SauceLabs.prototype.send = function (message, callback) {
   var method  = message.method,
       path    = message.path,
       args    = message.args,
       query   = message.query,
       data    = message.data,
-      body    = JSON.stringify(data);
+      body    = JSON.stringify(data),
+      uploadData  = message.uploadData;
```

Comment:
hmmm I don't like this much, can we not check if data = file buffer? Or something like that?